cmd/wallet-sdk-gomobile/didkey/didkey.go

```diff
@@ -0,0 +1,62 @@
+/*
+Copyright Gen Digital Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+// Package didkey contains a function that can be used to create did:key documents.
+package didkey
+
+import (
+	"errors"
+
+	"github.com/trustbloc/did-go/doc/did"
+	"github.com/trustbloc/wallet-sdk/pkg/walleterror"
+
+	"github.com/trustbloc/wallet-sdk/cmd/wallet-sdk-gomobile/api"
+	"github.com/trustbloc/wallet-sdk/cmd/wallet-sdk-gomobile/wrapper"
+	"github.com/trustbloc/wallet-sdk/pkg/did/creator/key"
+)
+
+// ErrorModule is the error module name for this package.
+const ErrorModule = "DIDKEY"
+
+// Create creates a new did:key document using the given JWK.
+func Create(jwk *api.JSONWebKey) (*api.DIDDocResolution, error) {
+	if jwk == nil || jwk.JWK == nil {
+		return nil, wrapper.ToMobileError(walleterror.NewInvalidSDKUsageError(
+			ErrorModule, errors.New("jwk object cannot be null/nil/empty")))
+	}
+
+	var vm *did.VerificationMethod
+
+	if jwk.JWK.Crv == "Ed25519" {
+		// Workaround: when the did:key VDR creates a DID for ed25519, Ed25519VerificationKey2018 is the expected
+		// verification method.
+		publicKeyBytes, err := jwk.JWK.PublicKeyBytes()
+		if err != nil {
+			return nil, wrapper.ToMobileError(err)
```

Comment:
Should these be in go pkg rather than mobile binding pkg ?

Reply:
I was also wondering the same thing. I wasn't sure if we wanted this behaviour to be common to all bindings that use the Go SDK or if it should be specific to the binding implementation (gomobile, WASM). Since you were wondering the same thing, I think perhaps it makes more sense for it to be common, so I'll update the PR.